enclave.go

```diff
@@ -81,12 +83,21 @@ type Config struct {
 	// is required.
 	FQDN string
 
-	// ExtPort contains the VSOCK-facing TCP port that the Web server should
+	// ExtPort contains the TCP port that the Web server should
 	// listen on, e.g. 443.  This port is not *directly* reachable by the
 	// Internet but the EC2 host's proxy *does* forward Internet traffic to
 	// this port.  This field is required.
 	ExtPort uint16
 
+	// UseVsockForExtPort must be set to true if direct communication
+	// between the host and Web server via VSOCK is desired. The daemon will listen
+	// on the enclave's VSOCK address and the port defined in ExtPort.
+	UseVsockForExtPort bool
```

Comment:
It's clever to bind the public-facing Web server directly to the VSOCK interface! That means we don't have to deal with the detour of sending packets to the tap device, to the kernel, and back to nitriding's TCP port.

Come to think of it, perhaps we don't need a toggle for this, and it should instead be the default behavior? It would mean however that we need two proxy functionalities on the host: one that encapsulates traffic (gvproxy) and one that simply forwards TCP streams (viproxy or vsock-relay). I think the additional proxy complexity on the host is worth the performance boost.

Reply:
my rationale for adding a switch is for backwards compatibility. if you're not concerned about that, then perhaps we could make this default behavior and add a switch for listing on ipv4/ipv6. your call!

Reply:
I think it's fine to have this toggle-able for now. We may later realize that we don't need this switch, at which point we can remove it, to keep things simple.